clients/src/main/java/org/apache/kafka/common/internals/Plugin.java

```diff
@@ -40,14 +47,46 @@ private Plugin(T instance, PluginMetricsImpl pluginMetrics) {
         this.pluginMetrics = Optional.ofNullable(pluginMetrics);
     }
 
+    /**
+     * Wrap an instance into a Plugin.
+     * @param instance the instance to wrap
```

Comment:
ditto I think we should also mention that the instance's class name will be used as the value for the `<code>class</code>` tag.